recipes/default.rb

```diff
@@ -46,6 +46,11 @@
   end
 end
 
+##  Set syncrepl_config dynamic values here
+node.default['openldap']['syncrepl_config']['searchbase'] = "\"#{node['openldap']['basedn']}\""
+node.default['openldap']['syncrepl_config']['binddn'] = "\"#{node['openldap']['syncrepl_cn']},#{node['openldap']['basedn']}\""
+node.default['openldap']['syncrepl_config']['credentials'] = "\"#{node['openldap']['slapd_replpw']}\""
```

Comment:
thats not a particularly useful use of attributes.  if anyone sets one of these they just get blown away here.  did you mean to use default_unless?

at any rate you should not set node attributes in recipe code and should construct a plain old ruby variable:

```ruby
synchrepl_config_searchbase = node['openldap']['syncrepl_config']['searchbase'] || %Q{"#{node['openldap']['basedn']}"}
```

and then inject that ruby variable into templates, etc.

this may complicate the template variable injection code, but you've headed down the path of abusing attributes here and its vastly better to stop doing that than to continue for the code simplicity in the templates.

you can construct a hash to pass to templates and add these variables and then merge the attributes if you do it carefully enough so that the attributes do not get modified.